chain-tx-enclave/enclave-utils/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg(all(feature = "sgxstd", target_env = "sgx"))]
+use aead::{generic_array::GenericArray, Aead, NewAead};
+#[cfg(all(feature = "sgxstd", target_env = "sgx"))]
+use aes_gcm::Aes128Gcm;
+use aes_gcm::Tag;
+#[cfg(all(feature = "sgxstd", target_env = "sgx"))]
+use sgx_isa::ErrorCode;
+use sgx_isa::Keyrequest;
+use std::convert::TryFrom;
+#[cfg(all(feature = "sgxstd", target_env = "sgx"))]
+use zeroize::Zeroize;
+
+/// Cryptographic payload in the sealed log of Intel SGX SDK
+/// (tag is detached, unlike in RFC-5116)
+pub struct AesGcmData {
+    pub payload_tag: Tag,
+    pub encrypt_txt: Vec<u8>,
+    pub additional_txt: Vec<u8>,
+}
+
+/// Data contained in the sealed log payload from Intel SGX SDK
+pub struct SealedData {
+    pub key_request: Keyrequest,
+    pub aes_data: AesGcmData,
+}
+
+impl SealedData {
+    pub fn try_copy_from(source: &[u8]) -> Option<Self> {
+        let mut pos: usize = 0;
+        let mut take = |n: usize| -> Option<&[u8]> {
```

Comment:
Can we return a `Result` from this closure? Maybe create a custom `Error` type with variant `Error::UnexpectedEndOfData`?

Reply:
I think `Option` is simpler and good enough when there is only one type of error.

Reply:
I tried to match what's in sgx-isa: https://edp.fortanix.com/docs/api/sgx_isa/struct.Keyrequest.html#method.try_copy_from

Reply:
Ok

Reply:
The reason I wanted to have an error type (`Parsing`) is that we can use that same error type while parsing other SGX stuff which can return error (like 'DerParsingError', etc.).